pynxtools_xps/vms/vamas.py

```diff
@@ -162,23 +180,47 @@ def _update_xps_dict_with_spectrum(self, spectrum, key_map):
             "energydispersion": f"{analyser_parent}/energydispersion",
             "detector": f"{analyser_parent}/detector",
             "manipulator": f"{instrument_parent}/manipulator",
-            "calibration": f"{instrument_parent}/calibration",
             "sample": f"{region_parent}/sample",
             "data": f"{region_parent}/data",
+            "energy_referencing": f"{region_parent}/calibrations/energy_referencing",
+            "peak_fitting": f"{region_parent}/peak_fitting",
             "region": f"{region_parent}",
         }
 
+        used_keys = []
+
         for grouping, spectrum_keys in key_map.items():
             root = path_map[str(grouping)]
             for spectrum_key in spectrum_keys:
-                try:
-                    units = re.search(r"\[([A-Za-z0-9_]+)\]", spectrum_key).group(1)
-                    mpes_key = spectrum_key.rsplit(" ", 1)[0]
+                mpes_key = spectrum_key.rsplit(" ", 1)[0]
+                self._xps_dict[f"{root}/{mpes_key}"] = spectrum[spectrum_key]
+
+                unit_key = f"{grouping}/{spectrum_key}"
+                units = self._get_units_for_key(unit_key)
+                if units:
                     self._xps_dict[f"{root}/{mpes_key}/@units"] = units
-                    self._xps_dict[f"{root}/{mpes_key}"] = spectrum[spectrum_key]
-                except AttributeError:
-                    mpes_key = spectrum_key
-                    self._xps_dict[f"{root}/{mpes_key}"] = spectrum[spectrum_key]
+
+        # Write process data
+        process_key_map: Dict[str, List[str]] = {
+            "energy_referencing": ["alignments"],
+            "peak_fitting": ["regions", "components"],
+        }
+
+        for grouping, process_key_list in process_key_map.items():
+            root = path_map[str(grouping)]
+            for spectrum_key in process_key_list:
+                try:
```

Comment:
No need to convert into `str` as you already maping `process_key_map` from `str` to `list`. is needed can be check is `str` or other types.

Reply:
removed the string conversion.